iot/api-client/beta-features/commands/send/send_test.py

```diff
@@ -0,0 +1,187 @@
+
+    topic = pubsub_client.create_topic(topic_path)
+    policy = pubsub_client.get_iam_policy(topic_path)
+
+    policy.bindings.add(
+        role='roles/pubsub.publisher',
+        members=['serviceAccount:cloud-iot@system.gserviceaccount.com'])
+
+    pubsub_client.set_iam_policy(topic_path, policy)
+
+    return topic
+
+
+def create_registry(
+        service_account_json, project_id, cloud_region, pubsub_topic,
+        registry_id):
+    """ Creates a registry and returns the result. Returns an empty result if
+    the registry already exists."""
+    client = send.get_client(service_account_json)
+    registry_parent = 'projects/{}/locations/{}'.format(
+            project_id,
+            cloud_region)
+    body = {
+        'eventNotificationConfigs': [{
+            'pubsubTopicName': pubsub_topic
+        }],
+        'id': registry_id
+    }
+    request = client.projects().locations().registries().create(
+        parent=registry_parent, body=body)
+
+    response = request.execute()
+    print('Created registry')
+    return response
+
+
+def delete_registry(
+       service_account_json, project_id, cloud_region, registry_id):
+    """Deletes the specified registry."""
+    print('Delete registry')
+    client = send.get_client(service_account_json)
+    registry_name = 'projects/{}/locations/{}/registries/{}'.format(
+            project_id, cloud_region, registry_id)
+
+    registries = client.projects().locations().registries()
+    return registries.delete(name=registry_name).execute()
+
+
+def create_device(
+        service_account_json, project_id, cloud_region, registry_id,
+        device_id, certificate_file):
+    """Create a new device without authentication."""
+    registry_name = 'projects/{}/locations/{}/registries/{}'.format(
+            project_id, cloud_region, registry_id)
+
+    with io.open(certificate_file) as f:
+        certificate = f.read()
+
+    client = send.get_client(service_account_json)
+    device_template = {
+        'id': device_id,
+        'credentials': [{
+            'publicKey': {
+                'format': 'RSA_X509_PEM',
+                'key': certificate
+            }
+        }]
+    }
+
+    devices = client.projects().locations().registries().devices()
+    return devices.create(parent=registry_name, body=device_template).execute()
+
+
+def delete_device(
+        service_account_json, project_id, cloud_region, registry_id,
+        device_id):
+    """Delete the device with the given id."""
+    print('Delete device')
+    client = send.get_client(service_account_json)
+    registry_name = 'projects/{}/locations/{}/registries/{}'.format(
+            project_id, cloud_region, registry_id)
+
+    device_name = '{}/devices/{}'.format(registry_name, device_id)
+
+    devices = client.projects().locations().registries().devices()
+    return devices.delete(name=device_name).execute()
+
+
+# Keep scaffolding from here
+@pytest.fixture(scope='module')
+def test_topic():
```

Comment:
Ditto on fixture name. Note: you can put all this stuff in a shared conftest.py file if it's identical to the logic in the other file.

Reply:
Done.